src/Chapter4.hs

```diff
@@ -673,6 +696,57 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+-- a polymorphic binary tree type
+data Tree a
+  = EmptyTree
+  | Node a (Tree a) (Tree a)
+  deriving (Show, Eq)
+
+-- insert new element in a tree
+insertElement :: Ord a => a -> Tree a -> Tree a
+insertElement x EmptyTree = Node x EmptyTree EmptyTree
+insertElement y (Node x left right)
+    | x < y = Node x (insertElement y left) right
+    | x > y = Node x left (insertElement y right)
+    | x == y = Node x left right
```

Comment:
It's generally preferable to use a single call to the `compare` function and pattern match on the result of this function exhaustively :slightly_smiling_face: This is a safer and faster approach :top: